### DISCUSSION TITLE
camera permission

### DISCUSSION THREAD

**Discussion by Ehsanwww** [j20001970/GDMP#29] [General]
Hello @j20001970 , for camera permission, it always return true, although camera is not plugged, how do you find camera is plugged and how do you find user have permission or not in godot ?

**Comment by j20001970** [comment]
Camera on desktop uses OpenCV for capturing webcam frames, which has no API for querying whether camera permission is granted, thats why calling `MediaPipeCameraHelper::permission_granted()` always return true on Linux/Windows, even if no camera device is plugged, and `request_permission()` is a stub method on desktop. If permission checking or camera device enumeration were to implemented on desktop, OS-specific calls would be required AFAIK.
If you call the same method on Android/iOS, corresponding API will be used to check if camera permission is actually granted.

**Reply by Ehsanwww** [reply]
@j20001970  thanks for answer, I found a solution, there is simple solution for that when using opencv, must use isOpened method of opencv
